rs/ic_os/config/src/lib.rs

```diff
@@ -31,8 +74,170 @@ fn parse_config_line(line: &str) -> Option<(String, String)> {
 pub fn config_map_from_path(config_file_path: &Path) -> Result<ConfigMap> {
     let file_contents = read_to_string(config_file_path)
         .with_context(|| format!("Error reading file: {}", config_file_path.display()))?;
-    Ok(file_contents
+
+    let normalized_file_contents = file_contents.replace("\r\n", "\n").replace("\r", "\n");
+
+    Ok(normalized_file_contents
         .lines()
         .filter_map(parse_config_line)
         .collect())
 }
+
+fn is_valid_ipv6_prefix(ipv6_prefix: &str) -> bool {
+    ipv6_prefix.len() <= 19 && format!("{ipv6_prefix}::").parse::<Ipv6Addr>().is_ok()
+}
+
+pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettings, bool)> {
```

Comment:
I've added unit test with the function as it currently stands. WDYT? Do you think I should still update it to take a reader?

Reply:
Yes it should take a reader.

EDIT: I'm not so sure now.  The tests are fine, and the functions can deal with file read errors there.

Reply:
Since setting up the test is not too difficult, I'm also fine with Paths. If we need to read configs from other sources in the future, we can still refactor it to `impl Read`